### ISSUE TITLE
error: could not build Objective-C module 'Realm'

### ISSUE THREAD

**Issue by i-schuetz** [realm/realm-cocoa#4429]
Hi, 

I was friendly told in https://github.com/realm/realm-cocoa/issues/1280 to open a new issue, since that one was already closed (it doesn't appear to be solved).

## Goals

Use realm in a framework target

## Expected Results

Compiles

## Actual Results

Doesn't compile:

/xxxxxx/Pods/Realm/Realm/module.modulemap:2:21: Umbrella header 'Realm.h' not found
Could not build Objective-C module 'Realm'

## Steps to Reproduce

Podfile:

```ruby
use_frameworks!

#framework
target 'Providers' do
    pod 'QorumLogs'
    pod 'Alamofire', '~> 4.0'
    pod 'Valet'
    pod 'RealmSwift'
    pod 'Reachability', '~> 3.2'
    pod 'Starscream'
end

target 'my_ios_target' do
    platform :ios, '8.0'
    pod 'SwiftValidator', :git => 'https://github.com/i-schuetz/SwiftValidator.git', :branch => 'remove_delegate_callback'
    pod 'SwiftCharts', :git => 'https://github.com/i-schuetz/SwiftCharts.git'
    pod 'Google/SignIn'
    pod 'SwipeView', '~> 1.3.2'
    pod 'CMPopTipView', '~> 2.0'
    pod 'KLCPopup', '~> 1.0'
    pod 'ChameleonFramework/Swift', :git => 'https://github.com/ViccAlexander/Chameleon.git'
    pod 'HockeySDK'
    pod 'QorumLogs'
end

target 'my_osx_target' do
    platform :osx, '10.10'
    pod 'Nimble', '~> 5.0.0'
end

#performance
post_install do |installer|
    puts("Update debug pod settings to speed up build time")
    Dir.glob(File.join("Pods", "**", "Pods*{debug,Private}.xcconfig")).each do |file|
        File.open(file, 'a') { |f| f.puts "\nDEBUG_INFORMATION_FORMAT = dwarf" }
    end
end
```

## Version of Realm and Tooling

ProductName:	Mac OS X
ProductVersion:	10.12.1
BuildVersion:	16B2555

/Applications/Xcode.app/Contents/Developer
Xcode 8.1
Build version 8B62

/Users/xxxxxx/.rbenv/shims/pod
1.2.0.beta.1
Realm (2.1.1)
RealmSwift (2.1.1)

/bin/bash
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin16)

/usr/local/bin/carthage
0.18.1
(not in use here)

/usr/bin/git
git version 2.9.3 (Apple Git-75)

## Additional notes

- I'm moving part of my code to a framework (embedded). Before of this everything (same code, dependencies/versions, environment) was working.

- None of the solutions from https://github.com/realm/realm-cocoa/issues/1280 helped.

Thanks!

**Comment by jeffreybergier** [comment]
I have this issue as well but with a much simple pod file and Xcode project:

# Steps to Reproduce:
1. Create new Xcode project with macOS/Cocoa Application
2. Create new pod file with the following contents:
3. Run Pod update or pod install
4. Open the workspace
5. Select the macOS app scheme and build
```
project 'RealmPodTest.xcodeproj'

target 'RealmPodTest' do
  platform :osx, '10.12'
  use_frameworks!
  pod 'RealmSwift'
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings['SWIFT_VERSION'] = '3.0.2'
    end
  end
end
```

# Current workaround
1. Find all Realm header files and select them in the left are of xcode
2. In the right area of Xcode change their Target Membership permission from Project to Public
3. Compile again

**Comment by jpsim** [comment]
@jeffreybergier this is with CocoaPods 1.2.0.beta.1, Xcode 8.1 and Realm Swift 2.1.1 as well?

**Comment by jeffreybergier** [comment]
@jpsim 

Version 8.2 beta (8C30a)
Cocoapods 1.0.0
Realm, 2.1.1
RealmSwift, 2.1.1

**Comment by jeffreybergier** [comment]
Adding a screenshot of the error if that helps. @i-schuetz if this is a different error than you're getting, let me know and I'll open a new Issue and stop hijacking yours :)
<img width="1400" alt="screen shot 2016-12-09 at 12 09 21" src="https://cloud.githubusercontent.com/assets/7349616/21065157/9cc94496-be13-11e6-8386-3de855edde3f.png">

**Comment by i-schuetz** [comment]
@jeffreybergier yes, same issue!

**Comment by i-schuetz** [comment]
In my case moving Realm's header files to public didn't help. Also not setting "Always search user paths" to yes. Path in user header search paths seems to be correct ("${PODS_ROOT}/Realm/include"). Also adding this to the header search paths instead didn't help. Just trying out things.

**Comment by i-schuetz** [comment]
I moved them in the build phases... under "Headers" you can drag & drop all the headers from project to public. Impossible to miss any.

**Comment by i-schuetz** [comment]
@jeffreybergier Actually, there is a difference between my error and yours (your first screenshot). The framework with the error at the left in my case is the framework I'm trying to build instead of  RealmSwift. The rest is the same.

**Comment by i-schuetz** [comment]
Hmm. I created a new project, with the same configuration (1 framework that uses Realm and main project uses framework) and podfile contents and it works. Double checked that the Realm version is the same (2.1.1), headers are marked as "project", etc.

How can I debug this, I really don't want to have to move everything to a new project, it's quite a lot...

**Comment by jeffreybergier** [comment]
@i-schuetz I see. Ok. I'm going to open a new issue for the issue I am seeing. Unfortunately I don't know enough about your project to help debug. But best of luck.

**Comment by i-schuetz** [comment]
@jeffreybergier no worries, thanks! the question was directed to the general/maintainers. 

I'm getting the feeling that I'll have to re-create the project.

For now I reverted to my monolithic project, still works :)

**Comment by i-schuetz** [comment]
Could reproduce exactly the same issue with the same project, on a brand new hard drive with updated macOS version, Xcode...

@jeffreybergier's workaround also didn't work here.

My new specs:

ProductName:	Mac OS X
ProductVersion:	10.12.2
BuildVersion:	16C67

/Applications/Xcode.app/Contents/Developer
Xcode 8.2
Build version 8C38

/usr/local/bin/pod
1.1.1
Realm (2.1.1)
RealmSwift (2.1.1)

/bin/bash
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin16)


(not in use here)

/usr/bin/git
git version 2.10.1 (Apple Git-78)

**Comment by i-schuetz** [comment]
Final update: I got it working by using a separate project instead of only separate target.

For completeness, I didn't need any workarounds to get it working with this new configuration.

**Comment by kishikawakatsumi** [comment]
@jeffreybergier It doesn't reproduce in my env, CocoaPods 1.1.1, Xcode 8.2, Realm 2.1.1. Could you make sure again the issue reproduces?

**Comment by jeffreybergier** [comment]
@kishikawakatsumi I just tried it with cocoapods 1.1.1 and it worked fine. It must be an issue with 1.0.0. I suppose we can close the issue. Thanks for looking into it!

**Comment by pallzoltan** [comment]
I'm getting the same error as @jeffreybergier, running Cocoapods 1.1.1. My headers are set to Public, but that didn't change anything. I'm getting this error only after including RLMSupport.swift.
